x-pack/legacy/plugins/reporting/config.ts

```diff
@@ -31,6 +31,17 @@ export async function config(Joi: any) {
         .default(120000),
     }).default(),
     capture: Joi.object({
+      timeouts: Joi.object({
+        openUrl: Joi.number()
+          .integer()
+          .default(30000),
+        waitForElements: Joi.number()
+          .integer()
+          .default(30000),
+        renderComplete: Joi.number()
+          .integer()
+          .default(30000),
```

Comment:
Note: These get passed to Puppeteer `waitForSelector` calls. Puppeteer's default for these is 30 seconds internally.

Reply:
I think we do call `https://pptr.dev/#?product=Puppeteer&version=v2.1.1&show=api-pagesetdefaulttimeouttimeout` and pass in the overall job timeout, so we might be overriding the default timeouts for these APIS.

Reply:
Yup, we do it here: https://github.com/elastic/kibana/blob/master/x-pack/legacy/plugins/reporting/server/browsers/chromium/driver_factory/index.ts#L120. Maybe we just default these to the global job timeout in this case?

Reply:
++ If we're passing the timeout to each of the calls, then setting `page.setDefaultTimeout` doesn't make a lot of sense.

I don't think we should use the global job timeout, which is the "queue timeout" because once that timeout hits there is no buffer room to recover a completed screenshot for the job. Hitting the "queue timeout" should be avoided as much as possible.

We need to think about this... Should the `page.setDefaultTimeout` be the max of these 3 new timeouts? 

At any rate, `server/browsers/chromium/driver_factory/index.ts#L120` is no longer valid in this PR

Reply:
@joelgriffith wdyt about the changes in 926b694319 ?

Reply:
Yup, this makes sense: 

```
        // All navigation methods default to 30 seconds,
        // which can cause the job to fail even if we bump timeouts in	        // All waitFor methods have their own timeout config
        // the config. Help alleviate errors like	        page.setDefaultTimeout(this.captureConfig.timeouts.openUrl);```

Reply:
Individual methods can still override this, but I like setting it to the `openUrl` which is where I think the bulk of the operation is for PDF reports